src/PointwiseFunctions/AnalyticSolutions/Xcts/Flatness.hpp

```diff
@@ -0,0 +1,131 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <limits>
+#include <ostream>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Elliptic/Systems/Xcts/Tags.hpp"
+#include "NumericalAlgorithms/LinearOperators/PartialDerivatives.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "PointwiseFunctions/AnalyticSolutions/Xcts/AnalyticSolution.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace Xcts::Solutions {
+
+/// \cond
+template <typename Registrars>
+struct Flatness;
+
+namespace Registrars {
+struct Flatness {
+  template <typename Registrars>
+  using f = Solutions::Flatness<Registrars>;
+};
+}  // namespace Registrars
+/// \endcond
+
+/// Flat spacetime in general relativity. Useful as initial guess.
+template <typename Registrars = tmpl::list<Solutions::Registrars::Flatness>>
+class Flatness : public AnalyticSolution<Registrars> {
```

Comment:
I would call this Minkowski as there are flat spacetime solutions with non-unit lapse and non-zero shift

Reply:
Not entirely sure about the nomenclature, but isn't 'Minkowski' and 'flat' pretty much synonymous, i.e. Minkowski spacetime is just flat spacetime and it can of course be sliced by choosing any lapse and shift? In other words, the Minkowski metric is a Kronecker delta only in Cartesian coordinates. So no matter if this is named 'Minkowski' or 'flat', it always needs to make a choice of coordinates. I think if we ever wanted to choose different coordinates for the flat solution it should be built into this class, similar to how the `Xcts::Solutions::Schwarzschild` class takes the coordinates as an enum (`Isotropic`, `PainleveGullstrand` or `KerrSchildIsotropic` in the Schwarzschild case).